.pre-commit-config.yaml

```diff
@@ -0,0 +1,56 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.1.0
+  hooks:
+  - id: check-added-large-files
+  - id: check-toml
+  - id: check-yaml
+  - id: end-of-file-fixer
+  - id: trailing-whitespace
+- repo: https://github.com/psf/black
+  rev: 22.1.0
+  hooks:
+  - id: black
+# TODO: Remove comments when type-hinting is implemented
+#- repo: https://github.com/pre-commit/mirrors-mypy
+#  rev: v0.931
+#  hooks:
+#  - id: mypy
+#    exclude: ^(docs/|example-plugin/|tests/fixtures)
+# TODO: Remove comments when/if poetry is implemented, AND when/if pytest is implemented
+#- repo: local
+#  hooks:
+#  - id: pytest
+#    name: Check pytest unit tests pass
+#    entry: poetry run duty test
+#    pass_filenames: false
+#    language: system
+#    types: [python]
+- repo: https://github.com/macisamuele/language-formatters-pre-commit-hooks
+  rev: v2.2.0
+  hooks:
+  - id: pretty-format-toml
+    args: [--autofix]
+  - id: pretty-format-yaml
+    args: [--autofix]
+    exclude: .copier-answers.yml
```

Comment:
```suggestion
    exclude: .github/workflows/codeql-analysis.yml
```
We don't have a `.copier-answers.yml` file but we do have a cron action in the `codeql-analysis.yml` file that gets the `'` removed around asterisks which is not a valid suggestion.

As per [the doc here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onschedule)
`* is a special character in YAML so you have to quote this string`

<details>  
There are things in the `codeql-analysis.yml` file that should be dealt with but this can be dealt with another pr probably.
</details>